### PR TITLE
Deprecate tags.json

### PR DESCRIPTION
**Pull request by karsa-mistmere**
This PR will deprecate the use of tags.json

### CODE REVIEW

File: site/src/pages/packages/index.tsx [hunk 1]
```diff
@@ -1,65 +1,3 @@
-import Layout from '../../components/Layout';
```
[the remark this answers is not in the document]
Comment (karsa-mistmere) [reply site/src/pages/packages/index.tsx]: Yeah, definitely not, thanks.

**Review by ericfennis — APPROVED** [review]
Nice well done this looks good!

**Comment by jguddas** [comment]
What is left for this to get merged?

**Comment by karsa-mistmere** [comment]
> What is left for this to get merged?

We need to add caching to the API endpoints.

**Comment by jguddas** [comment]
Is it just `getData` or is there anything else that needs to be cached?
```ts
const iconDataCache = new Map<string, IconEntity>();
export async function getData(name: string) {
  if (iconDataCache.has(name)) return iconDataCache.get(name);
  const svgPath = path.join(directory, `${name}.svg`);
  const svgContent = fs.readFileSync(svgPath, 'utf8');
  const jsonPath = path.join(directory, `${name}.json`);
  const jsonContent = fs.readFileSync(jsonPath, 'utf8');
  const iconJson = JSON.parse(jsonContent) as Pick<IconEntity, 'tags'>;

  const contributors = await getContributors(name);

  const data = {
    ...iconJson,
    name,
    contributors,
    src: svgContent,
  };

  iconDataCache.set(name, data);
  return data;
}
```

**Comment by karsa-mistmere** [comment]
@jguddas: No, it's not about `getData`, we'll need to add cache control in Next.js to the new API endpoints for `/api/icon-nodes` and `/api/tags`.

**Comment by jguddas** [comment]
Ahh I see, why have this be an API route at all, tags and icon-nodes are static content and can be generated at build time?

**Comment by karsa-mistmere** [comment]
> Ahh I see, why have this be an API route at all, tags and icon-nodes are static content and can be generated at build time?

Yeah, currently we sure could, and I think an acceptable solution would be building the relevant data at build time and serving that, _but_: cache control in Next.js will still need to be added so requests are optimized.

**Comment by jguddas** [comment]
Sure caching is always nice, but why do we have those endpoints at all?
We currently use https://unpkg.com/lucide-static@latest/tags.json and that will still exist in the future right?

**Comment by karsa-mistmere** [comment]
I'm not quite sure it will, actually. 😅

**Comment by jguddas** [comment]
What the problem with the CDN, a CDN is the right choice for static assets like that, no?

**Comment by karsa-mistmere** [comment]
We're looking into the possibility of making it more dynamic in the future.

File: packages/lucide-figma/src/api/fetchIcons.ts [hunk 2]
```diff
@@ -23,8 +23,8 @@ export const fetchIcons = async (cachedIcons? : LucideIcons): Promise<LucideIcon
     return cachedIcons
   }
 
-  const iconNodesResponse = await fetch(`https://unpkg.com/lucide-static@${packageJson.version}/icon-nodes.json`)
-  const tagsResponse = await fetch('https://unpkg.com/lucide-static@latest/tags.json')
+  const iconNodesResponse = await fetch(`https://lucide.dev/api/icon-nodes`)
+  const tagsResponse = await fetch('https://lucide.dev/api/tags')
```
Comment (jguddas) [inline packages/lucide-figma/src/api/fetchIcons.ts:27]: I would only add those endpoints when they are needed and keep using the CDN for now :man_shrugging:.
```suggestion
  const [iconNodesResponse, tagsResponse] = await Promise.all([
    fetch(`https://unpkg.com/lucide-static@${packageJson.version}/icon-nodes.json`),
    fetch('https://unpkg.com/lucide-static@latest/tags.json')
  ])
```

File: packages/lucide-static/package.json [hunk 3]
```diff
@@ -12,11 +12,11 @@
   "main": "lib/index.js",
   "scripts": {
     "copy:icons": "cp -r ../../icons icons",
-    "copy:tags": "cp ../../tags.json tags.json",
     "copy:license": "cp ../../LICENSE ./LICENSE",
-    "build": "pnpm clean && pnpm copy:license && pnpm copy:icons && pnpm copy:tags && pnpm build:lib",
+    "build:tags": "node ../../scripts/migrateIconsToTags.mjs",
+    "build": "pnpm clean && pnpm copy:license && pnpm copy:icons && pnpm build:lib",
```
Comment (jguddas) [inline packages/lucide-static/package.json:17]:
```suggestion
    "build": "pnpm clean && pnpm copy:license && pnpm copy:icons && pnpm build:lib && pnpm build:tags",
```

File: site/src/pages/api/icon-nodes/index.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import {getAllData} from '../../../lib/icons';
+import { parseSync } from 'svgson';
```
Comment (jguddas) [inline site/src/pages/api/icon-nodes/index.tsx:2]: `svgson` is not in the `package.json`

Comment (karsa-mistmere) [reply site/src/pages/api/icon-nodes/index.tsx:2]: I don't think it needs to be with pnpm and the workspace package.json: https://github.com/lucide-icons/lucide/blob/main/package.json#L42

Feel free to add it though.

File: packages/lucide-figma/src/api/fetchIcons.ts (hunk 2 above)
Comment (karsa-mistmere) [reply packages/lucide-figma/src/api/fetchIcons.ts:27]: > I would only add those endpoints when they are needed and keep using the CDN for now 🤷‍♂️.

We've discussed this with the other maintainers and we intend to move forward with deprecating the CDN call and switching to an API.

Comment (jguddas) [reply packages/lucide-figma/src/api/fetchIcons.ts:27]: You could still put it in a `Promise.all` :+1:

Comment (karsa-mistmere) [inline packages/lucide-figma/src/api/fetchIcons.ts:27]:
```suggestion
  const [iconNodesResponse, tagsResponse] = await Promise.all([
    fetch('https://lucide.dev/api/icon-nodes'),
    fetch('https://lucide.dev/api/tags')
  ])
```

Comment (karsa-mistmere) [reply packages/lucide-figma/src/api/fetchIcons.ts]: Sure, that's a no brainer, thx.

**Review by jguddas — APPROVED** [review]

File: site/src/pages/api/icon-nodes/index.tsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+export default async function handler(req, res) {
+  const icons = await getAllData();
+  res.status(200).json(icons.reduce((acc, icon, i) => {
```
Comment (ericfennis) [inline site/src/pages/api/icon-nodes/index.tsx:6]:
```suggestion
  // Caching of one day
  res.setHeader('Cache-Control', 's-maxage=86400, public');
  res.status(200).json(icons.reduce((acc, icon, i) => {
```

File: site/src/pages/api/tags/index.tsx [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import {getAllData} from '../../../lib/icons';
+
+export default async function handler(req, res) {
+  const icons = await getAllData();
+  res.status(200).json(icons.reduce((acc, icon, i) => {
```
Comment (ericfennis) [inline site/src/pages/api/tags/index.tsx:5]:
```suggestion
  // Caching of one day
  res.setHeader('Cache-Control', 's-maxage=86400, public');
  res.status(200).json(icons.reduce((acc, icon, i) => {
```

**Review by ericfennis — COMMENTED** [review]
Here are the Cache control headers to cache the response in Vercel.

From the docs: https://vercel.com/docs/concepts/functions/serverless-functions/edge-caching

**Comment by ericfennis** [comment]
@karsa-mistmere I made sure script for tags.json is added to the lucide-static package.

**Review by ericfennis — APPROVED** [review]

**Comment by ericfennis** [comment]
@karsa-mistmere Hmm when testing on preview URL. I got a 500 error.
image
https://lucide-32l1n891o-lucide.vercel.app/api/tags
https://lucide-32l1n891o-lucide.vercel.app/api/icon-nodes

**Comment by ericfennis** [comment]
image
The Error I found in Vercel

File: site/src/lib/nextCache.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+  return null
+}
+
+const write = (cacheKey: string, content: any) => {
+  if (!fs.existsSync(cacheDir)) {
+    fs.mkdirSync(cacheDir)
+  }
+
+  fs.writeFileSync(cachePath(cacheKey), JSON.stringify(content), 'utf-8')
```
Comment (ericfennis) [inline site/src/lib/nextCache.tsx:21]: We can't use `fs.write` during runtime in Vercel.
What we can do is create a cache on build time. 
So we need to create a separate script we execute when the build command is fired.

**Comment by ericfennis** [comment]
Awesome work great now!

**Review by ericfennis — APPROVED** [review]

**Pull request merged**